src/mono/mono/metadata/metadata-update.c

```diff
@@ -814,11 +814,58 @@ apply_enclog_pass1 (MonoImage *image_base, MonoImage *image_dmeta, gconstpointer
 		int token_table = mono_metadata_token_table (log_token);
 		int token_index = mono_metadata_token_index (log_token);
 
-		if (token_table == MONO_TABLE_ASSEMBLYREF) {
+		switch (token_table) {
+		case MONO_TABLE_ASSEMBLYREF:
 			/* okay, supported */
-		} else if (token_table == MONO_TABLE_METHOD) {
+			break;
+		case MONO_TABLE_METHOD:
 			/* handled above */
-		} else {
+			break;
+		case MONO_TABLE_PROPERTY:
+			if (token_index <= table_info_get_rows (&image_base->tables [token_table])) {
+				/* modifying a property */
+				break;
+			} else {
```

Comment:
Purely stylistic, but I preferred the way you handled the conditional in the `METHOD_SEMANTIC_SETTER` case.